dali/operators/image/remap/warp_param_provider.h

```diff
@@ -311,31 +315,20 @@ class WarpParamProvider : public InterpTypeProvider, public BorderTypeProvider<B
 
   /** @brief Allocates num_samples_ MappingParams objects in memory specified by alloc  */
   template <typename MemoryKind>
-  MappingParams *AllocParams() {
-    return AllocParams<MemoryKind>(num_samples_);
-  }
-
-  template <typename MemoryKind>
-  auto &SelectParamView() {
-    return SelectParamView(static_cast<MemoryKind*>(nullptr));
+  MappingParams *AllocParams(int num_samples) {
+    return AllocParams(static_cast<MemoryKind*>(nullptr), num_samples);
   }
 
-  inline auto &SelectParamView(mm::memory_kind::device *) {
-    return params_gpu_;
+  inline MappingParams *AllocParams(mm::memory_kind::device *, int count) {
+    params_gpu_ = mm::alloc_raw_unique<MappingParams, mm::memory_kind::device>(count);
+    params_count_ = count;
+    return params_gpu_.get();
```

Comment:
If you keep the logic of storing the `params_xpu_` as members (which we probably should reconsider), then there's no point reallocating this buffer every time.
```suggestion
    if (count > params_count_) {
      params_gpu_ = mm::alloc_raw_unique<MappingParams, mm::memory_kind::device>(count);
      params_count_ = count;
    }
    return params_gpu_.get();
```